it/uk/gov/hmrc/agentepayeregistration/controllers/AgentEpayeRegistrationControllerISpec.scala

```diff
@@ -0,0 +1,84 @@
+package uk.gov.hmrc.agentepayeregistration.controllers
+
+import play.api.libs.json.Json
+import play.api.test.Helpers._
+import play.api.test.{FakeHeaders, FakeRequest}
+import uk.gov.hmrc.agentepayeregistration.repository.AgentEpayeRegistrationRepository
+
+class AgentEpayeRegistrationControllerISpec extends BaseControllerISpec {
+  private lazy val controller: AgentEpayeRegistrationController = app.injector.instanceOf[AgentEpayeRegistrationController]
+  private lazy val repo = app.injector.instanceOf[AgentEpayeRegistrationRepository]
+
+  "submitting valid details to /register" should {
+    "respond with HTTP 200 with a the new unique PAYE code in the response body" in {
+      val postData = Json.parse(
```

Comment:
postData can be refactored to only be declared once, with methods that remove or add fields. Can look up the methods if you want.